pkg/virtual/framework/wrappers/rbac/merging.go

```diff
@@ -0,0 +1,469 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbac
+
+import (
+	"context"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	rbacinformers "k8s.io/client-go/informers/rbac/v1"
+	rbaclisters "k8s.io/client-go/listers/rbac/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func MergedInformers(informers ...rbacinformers.Interface) rbacinformers.Interface {
+	return &mergedInterface{
+		informers: informers,
+	}
+}
+
+var _ rbacinformers.Interface = (*mergedInterface)(nil)
+
+type mergedInterface struct {
+	informers []rbacinformers.Interface
+}
+
+func (i *mergedInterface) ClusterRoleBindings() rbacinformers.ClusterRoleBindingInformer {
+	aggregatedInformers := make([]rbacinformers.ClusterRoleBindingInformer, 0)
+	for _, inf := range i.informers {
+		aggregatedInformers = append(aggregatedInformers, inf.ClusterRoleBindings())
+	}
+	return MergedClusterRoleBindingInformer(aggregatedInformers)
+}
+
+func (i *mergedInterface) ClusterRoles() rbacinformers.ClusterRoleInformer {
+	aggregatedInformers := make([]rbacinformers.ClusterRoleInformer, 0)
+	for _, inf := range i.informers {
+		aggregatedInformers = append(aggregatedInformers, inf.ClusterRoles())
+	}
+	return MergedClusterRoleInformer(aggregatedInformers)
+}
+
+func (i *mergedInterface) RoleBindings() rbacinformers.RoleBindingInformer {
+	aggregatedInformers := make([]rbacinformers.RoleBindingInformer, 0)
+	for _, inf := range i.informers {
+		aggregatedInformers = append(aggregatedInformers, inf.RoleBindings())
+	}
+	return MergedRoleBindingInformer(aggregatedInformers)
+}
+
+func (i *mergedInterface) Roles() rbacinformers.RoleInformer {
+	aggregatedInformers := make([]rbacinformers.RoleInformer, 0)
+	for _, inf := range i.informers {
+		aggregatedInformers = append(aggregatedInformers, inf.Roles())
+	}
+	return MergedRoleInformer(aggregatedInformers)
+}
+
+func MergedClusterRoleBindingInformer(informers []rbacinformers.ClusterRoleBindingInformer) rbacinformers.ClusterRoleBindingInformer {
+	return &mergedClusterRoleBindingInformer{
+		informers: informers,
+	}
+}
+
+var _ rbacinformers.ClusterRoleBindingInformer = (*mergedClusterRoleBindingInformer)(nil)
+var _ rbaclisters.ClusterRoleBindingLister = (*mergedClusterRoleBindingLister)(nil)
+
+type mergedClusterRoleBindingInformer struct {
+	informers []rbacinformers.ClusterRoleBindingInformer
+}
+
+type mergedClusterRoleBindingLister struct {
+	listers []rbaclisters.ClusterRoleBindingLister
+}
+
+func (i *mergedClusterRoleBindingInformer) Informer() cache.SharedIndexInformer {
+	panic("not implemented")
+}
+
+func (i *mergedClusterRoleBindingInformer) Lister() rbaclisters.ClusterRoleBindingLister {
```

Comment:
do we have to implement all the binding interfaces? Can we strip the whole file down to ClusterRoles and Roles?

Reply:
Done